### PR TITLE
feature: implemented ngx.resp.add_header.

### PR DESCRIPTION
**Pull request by spacewander**
The `ngx.resp.add_header` works like the `add_header` Nginx directive.
Unlike the `ngx.header.HEADER=`, this method appends new header to the old one
instead of overriding it.
Unlike the `add_header` directive, this method overrides the builtin header
instead of appending it. IMHO, it is reasonable since appending to the
builtin header like `Content-Length` is confusing.

TODO: add documentation.

This pr depends on https://github.com/openresty/meta-lua-nginx-module/pull/32.

I hereby granted the copyright of the changes in this pull request
to the authors of this lua-resty-core project.

### CODE REVIEW

File: lib/resty/core/response.lua [hunk 1]
```diff
@@ -52,8 +52,13 @@ local function set_resp_header(tb, key, value)
 
     local rc
     if value == nil then
-        rc = C.ngx_http_lua_ffi_set_resp_header(r, key, #key, true, nil, 0,
-                                                nil, 0, errmsg)
+        if no_override then
+            error("invalid header value", 3)
+        else
```
Comment (doujiang24) [inline lib/resty/core/response.lua]: we can save the `else` here.

Comment (agentzh) [reply lib/resty/core/response.lua]: Agreed.

**Comment by spacewander** [comment]
@agentzh 
Updated.

File: t/response.t [hunk 2]
```diff
@@ -9,7 +9,7 @@ use Cwd qw(cwd);
```
Comment (agentzh) [inline t/response.t]: We can just use shorter test file names like `resp.t` here? ;)

File: lib/resty/core/response.lua [hunk 3]
```diff
@@ -160,6 +165,11 @@ do
 end
 
 
+function ngx.resp.add_header(key, value)
```
Comment (agentzh) [inline lib/resty/core/response.lua]: Let's introduce an `ngx.resp` Lua module which can be loaded on user's demand.

File: lib/resty/core/response.lua [hunk 4]
```diff
@@ -31,7 +31,7 @@ ffi.cdef[[
     int ngx_http_lua_ffi_set_resp_header(ngx_http_request_t *r,
         const char *key_data, size_t key_len, int is_nil,
         const char *sval, size_t sval_len, ngx_http_lua_ffi_str_t *mvals,
-        size_t mvals_len, char **errmsg);
+        size_t mvals_len, unsigned override, char **errmsg);
```
Comment (agentzh) [inline lib/resty/core/response.lua]: Please use `int` instead of `unsigned` for the `override` parameter.

**Comment by agentzh** [comment]
Also, will you please add some docs for this new API function to the `lib/ngx/resp.md` file and add a link to README? Thanks!

**Comment by spacewander** [comment]
@agentzh 
Have been moved to `lib/ngx/resp`.

**Pull request closed without merging**